### DISCUSSION TITLE
[OP Stack Deployment] Configure P2P Sequencer Address

### DISCUSSION THREAD

**Discussion by con5cience** [ethereum-optimism/developers#624] [Deploying the OP Stack] [answered]
### Did you check the documentation?

- [X] Yes

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

Super simple question within a very complex context.

I've built a devnet sequencer on Optimism per the "Create Your Own Rollup" docs. I'm now trying to stand up a Reth + Optimism node to peer with the network because there are technical dependencies that only Reth satisfies. I'm seeing this error in the Optimism logs on that peer:
```"no configured p2p sequencer address, ignoring gossiped block"```

Where/how do I configure a P2P Sequencer Address? I've tried reverse-engineering the codebase, it seems to refer to a parameter in the rollup configuration that doesn't exist.

Any help would be appreciated. 🙏

**Comment by hash1go** [comment]
Hello.
I think the sequencer address refers to this address.
https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup#generate-addresses
> The Sequencer address signs blocks on the p2p network.

**Reply by con5cience** [reply]
Hi there!

Thanks for your comment.

I definitely understand the notion of the sequencer address -- What I don't get is how or where to actually hand off the address to `op-node`.

Let me know if that helps clarify things.

**Reply by hash1go** [reply]
With this procedure, information about the sequencer address is registered as a host environment variable, so the sequencer's op-node should be able to reference it.
https://docs.optimism.io/builders/chain-operators/tutorials/create-l2-rollup#load-environment-variables

Additionally, the sequencer address becomes accessible through the L1 contract (SystemConfig).
https://etherscan.io/address/0x229047fed2591dbec1eF1118d64F7aF3dB9EB290#readProxyContract#F27

**Reply by con5cience** [reply]
Right, but this is an `op-node` process not running on the sequencer itself -- does `op-node` consume that environment variable regardless of whether it's running in Sequencer mode?

Since I have to hand off the rollup configuration to `op-node` for my custom devnet peer, does that somehow imply that the sequencer address didn't make it into that file? How would I confirm, and once confirmed, how do I correct it?

**Reply by hash1go** [reply, marked as answer]
Sorry, I thought you were talking about op-node on the sequencer.
The L1 system config contract address should be written in `rollup.json`. For more details, please check this troubleshooting guide.

https://github.com/ethereum-optimism/developers/discussions/17#discussioncomment-9424817

**Reply by con5cience** [reply]
I had unintentionally included the `l1.beacon` flag and also didn't have the `p2p` flags on my `op-node` init script as per the supplied link.

The link was exceptionally helpful, I have a devnet peer syncing now. Thank you so much for your help! <3
